src/utils/Cypher.ts

```diff
@@ -26,19 +34,168 @@ export default class Cypher {
     return this.configuration;
   }
 
-  public static encrypt(data: string): string {
+  public static async migrateCryptoKey(tenantID: string): Promise<void> {
+    // Crypto Key from config file
+    const configCryptoKey: string = Configuration.getCryptoConfig().key;
+
+    // Crypto Key Setting from db
+    const keySettings = await SettingStorage.getCryptoSettings(tenantID);
+
+    // If no Crypto Key Setting exist, initialize them with Crypto Key from config file
+    if (!keySettings) {
+      // Create New Crypto Key in Tenant Settings
+      const keySettingToSave = {
+        identifier: TenantComponents.CRYPTO,
+        type: CryptoSettingsType.CRYPTO,
+        crypto: {
+          key: configCryptoKey,
+          migrationDone: true
+        }
+      } as KeySettings;
+      await SettingStorage.saveCryptoSettings(tenantID, keySettingToSave);
+    }
+  }
+
+  public static getCryptoKeySync(): string {
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
+  }
+
+  public static async getCryptoKey(tenantID: string): Promise<string> {
+    if (!this.cryptoSetting) {
+      this.keySetting = await SettingStorage.getCryptoSettings(tenantID);
+      this.cryptoSetting = this.keySetting.crypto;
+    }
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
```

Comment:
You can also use the helper method here.

Reply:
which helper method you mean?

Reply:
The one to get the correct crypto key.

Reply:
But this is the function which gets the correct crypto key. I created the getCryptoKeySync() only because in StripeBillingIntegration.ts I need to get the key in the constructor and cannot use the asyncronous getCryptoKey()

Reply:
> I would expect that you first try to access the DB and next that you fallback to the value in the config.json file.
> Do not access 'this' in a static method!
> Moreover 'cryptoSetting' is per tenant, why do you keep it as a member class variable!?

Given the review, making the Cypher class multi tenant aware should solve most of the comments. 

@stefania-santimbrean : Cypher can stay mostly has an helpers class of static methods but the all public methods shall be tenant aware.